### PR TITLE
feat: add helm deploy workflow to deploy app to aks cluster

### PR DESCRIPTION
**Pull request by christopherswarren**
Adds helm deploy workflow job to send app to cluster
Adds db connection check
Reverts Dockerfile to working state
Added step to create namespace if it does not exist
Updated to use azure/aks-set-context@v1 action for cluster authentication
Updates dependencies according to Dependabot findings
Adds tested tls-cert configuration to helm chart

### CODE REVIEW

File: .github/workflows/post-release.yml [hunk 1]
```diff
@@ -42,3 +42,49 @@ jobs:
...
+        with:
+          fetch-depth: 0
+
+      - name: AZ Login
+        env:
+          SERVICE_PRINCIPAL_APP_ID: ${{ secrets.CLUSTER_CRED_GETTER_APPID }}
+          SERVICE_PRINCIPAL_SECRET: ${{ secrets.CLUSTER_CRED_GETTER_PWD }}
+          SERVICE_PRINCIPAL_TENANT: ${{ secrets.CLUSTER_TENANTID }}
+        run: |
+          az login \
```
Comment (mrparkers) [inline .github/workflows/post-release.yml:69]: maybe we should use this instead?  https://github.com/Azure/login

File: Dockerfile [hunk 2]
```diff
@@ -13,18 +13,16 @@ COPY routes /envelope-game/routes
...
 COPY __mocks__ /envelope-game/__mocks__
 
-RUN yarn test
+RUN npm test
 
-RUN yarn run build
+RUN npm run build
 
-RUN yarn install --production --ignore-scripts --prefer-offline
+# RUN npm prune --production
```
Comment (djtjwillia) [inline Dockerfile:20]: if we don't need to do this, we should remove this line.

Comment (christopherswarren) [reply Dockerfile:20]: agreed, removed

File: Dockerfile [hunk 3]
```diff
@@ -1,10 +1,10 @@
-FROM node:16-alpine AS test
+FROM node:14-alpine AS test
 
 WORKDIR /envelope-game
 
-COPY package*.json yarn.lock /envelope-game/
+COPY package*.json /envelope-game/
```
[the remark this answers is not in the document]
Comment (christopherswarren) [reply Dockerfile]: Talked further with Parker and un-did these changes related to npm/yarn

File: .github/workflows/post-release.yml (hunk 1 above)
Comment (christopherswarren) [reply .github/workflows/post-release.yml:63]: converted to use this action

File: .github/workflows/post-release.yml [hunk 4]
```diff
@@ -33,12 +37,37 @@ jobs:
         uses: docker/login-action@v1
         with:
           registry: ghcr.io
-          username: ${{ secrets.CR_USER }}
```
Comment (mrparkers) [inline .github/workflows/post-release.yml:36]: why was this removed?

File: charts/envelope-game/templates/deployment.yaml [hunk 5]
```diff
@@ -54,11 +50,12 @@ spec:
           livenessProbe:
             httpGet:
               path: /
-              port: http
+              port: 3000
```
Comment (mrparkers) [inline charts/envelope-game/templates/deployment.yaml:53]: we probably don't need to change this.  it's nice to reference a port by its name so we only have to change it in a single place.

Comment (christopherswarren) [reply charts/envelope-game/templates/deployment.yaml:53]: reverting these, thanks!

File: .github/workflows/post-release.yml (hunk 4 above)
Comment (christopherswarren) [reply .github/workflows/post-release.yml:36]: confusion about possibly not needing creds for public repos/artifacts... replacing!

File: package.json [hunk 6]
```diff
@@ -10,17 +10,27 @@
     "@fortawesome/fontawesome-svg-core": "^1.2.30",
     "@fortawesome/free-solid-svg-icons": "^5.14.0",
     "@fortawesome/react-fontawesome": "^0.1.11",
+    "ansi-regex": "^3.0.1",
```
Comment (mrparkers) [inline package.json:13]: why are all of these dependencies being added?

File: charts/envelope-game/templates/certificate.yaml [hunk 7]
```diff
@@ -6,17 +6,22 @@ metadata:
 spec:
   acme:
     email: cloudservices@liatr.io
-    server: https://acme-v02.api.letsencrypt.org/directory
+    server: https://acme-staging-v02.api.letsencrypt.org/directory
```
Comment (mrparkers) [inline charts/envelope-game/templates/certificate.yaml:9]: maybe this should be a value so we don't have to update the template every time we want to change this

File: charts/envelope-game/values.yaml [hunk 8]
```diff
@@ -43,13 +51,14 @@ ingress:
     - secretName: envelopegame-cert
       hosts:
         - envelopegame.com
+        - envelope-game.aks-lz-dev.azurecaf.liatr.io
```
[the remark this answers is not in the document]
Comment (christopherswarren) [reply charts/envelope-game/values.yaml]: Updated to no longer issue cert for both domains. Will use cluster wildcard for azure long-form domain.

**Review by mrparkers — APPROVED** [review]
